### PR TITLE
Document search channel endpoints

### PR DESCRIPTION
**Pull request by krauselukas**
*TODO*:

Document...

- [x] search/channel
- [x] search/channel/binary
- [x] search/channel/binary/id

### CODE REVIEW

**Comment by krauselukas** [comment]
[Search with "match" parameter (xpath) (L)](https://trello.com/c/GlMt3MXk/1804-search-with-match-parameter-xpath-l)

File: src/api/public/apidocs-new/OBS-v2.10.50.yaml [hunk 1]
```diff
@@ -231,6 +231,12 @@ paths:
     $ref: 'paths/search_request.yaml'
   /search/request/id:
     $ref: 'paths/search_request_id.yaml'
+  /search/channel:
+    $ref: 'paths/search_channel.yaml'
+  /search/channel/binary:
+    $ref: 'paths/search_channel_binary.yaml'
+  /search/channel/binary/id:
+    $ref: 'paths/search_channel_binary_id.yaml'
```
Comment (eduardoj) [inline src/api/public/apidocs-new/OBS-v2.10.50.yaml:239]: We should keep the endpoints alphabetically orderer.These should go between the `/search` and `/search/issue` endpoints.

File: src/api/public/apidocs-new/paths/search_channel.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  parameters:
+    - in: query
+      name: match
+      schema:
+        type: string
+      required: yes
+      description: |
+        Expression based in XPath.
+        Providing a value of `*` will return all channels.
+        Available predicates are:
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/search_channel.yaml:16]:
```suggestion
        Expression based in XPath.

        Providing a value of `*` will return all channels.

        Available predicates are: 
```
Without adding a blank line in between all text is rendered in the same line. I find it more clear to split the text in different paragraphs.

Screenshot from 2022-08-25 16-53-14

File: src/api/public/apidocs-new/paths/search_channel.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        application/xml; charset=utf-8:
+          schema:
+            $ref: '../components/schemas/channel/collection.yaml'
+          examples:
+            no_results:
+              summary: No matching results
+              value:
+                matches: 0
+            two_results:
+              summary: One matching results
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/search_channel.yaml:43]:
```suggestion
              summary: One matching result
```

File: src/api/public/apidocs-new/paths/search_channel_binary_id.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
+get:
+  summary: List channel objects that match a XPath condition.
+  description: |
+    Return a collection of channel objects that match a XPath condition.
+
+    This operation is the same as the one defined with `GET /search/channel/binary/id` with the exception of the results returned.
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/search_channel_binary_id.yaml:6]:
```suggestion
    This operation is the same as the one defined with `GET /search/channel/binary` with the exception of the results returned.
```

Comment (krauselukas) [reply src/api/public/apidocs-new/paths/search_channel_binary_id.yaml]: done

File: src/api/public/apidocs-new/paths/search_channel.yaml (hunk 3 above)
Comment (krauselukas) [reply src/api/public/apidocs-new/paths/search_channel.yaml]: done

File: src/api/public/apidocs-new/OBS-v2.10.50.yaml (hunk 1 above)
Comment (krauselukas) [reply src/api/public/apidocs-new/OBS-v2.10.50.yaml]: done

File: src/api/public/apidocs-new/paths/search_channel_binary_id.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    objects with the channel definition related to the binary.
+    See example values of a succeeded request below.
+  security:
+    - basic_authentication: []
+  parameters:
+    - in: query
+      name: match
+      schema:
+        type: string
+      required: yes
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/search_channel_binary_id.yaml:17]:
```suggestion
```
In this endpoint `match` parameter is not required.

File: src/api/public/apidocs-new/paths/search_channel_binary_id.yaml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      description: |
+        Expression based in XPath.
+
+        Providing a value of `*` will return all channels.
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/search_channel_binary_id.yaml:21]:
```suggestion
        Not providing a value or providing a value of `*` will return all released binaries.
```
In this case it is allowed not providing a value.

File: src/api/public/apidocs-new/paths/search_channel_binary.yaml [hunk 7]
```diff
@@ -0,0 +1,85 @@
+get:
+  summary: List channel objects that match a XPath condition.
+  description: |
+    Return a collection of channel objects with release informations that match a XPath condition.
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/search_channel_binary.yaml:4]:
```suggestion
    Return a collection of channel objects with release informations that match a XPath condition.
    
     This operation is the same as the one defined with `GET /search/channel/binary/id` with the exception of the results returned.
     While the former operation only returns a list of channel objects with the channel definition related to the binary, this one returns a list of channel
     objects with release information.
     See example values of a succeeded request below.
```
The reference to the `GET /search/channel/binary/id` was missing.

**Review by eduardoj — APPROVED** [review]

**Comment by hennevogel** [comment]
codecov hangs. [coverage is fine](https://app.circleci.com/pipelines/github/krauselukas/open-build-service/496/workflows/f04a1119-ac09-4bd6-92b7-36538735236e/jobs/3847), merging.

**Pull request merged**